### ISSUE TITLE
IOS-XR Fails to parse "show l2vpn bridge-domain"

### ISSUE THREAD

**Issue by pyhas** [CiscoTestAutomation/genielibs#168]
getting the below traceback when running genie

Traceback (most recent call last):
  File "/home/nmsadmin/scripts/infograb/pyats_genie.py", line 15, in <module>
    show_l2vpn = iosxr1.parse('show l2vpn bridge-domain')
  File "src/genie/conf/base/device.py", line 529, in genie.conf.base.device.Device.parse
  File "src/genie/conf/base/device.py", line 568, in genie.conf.base.device.Device._get_parser_output
  File "src/genie/conf/base/device.py", line 566, in genie.conf.base.device.Device._get_parser_output
  File "src/genie/metaparser/_metaparser.py", line 329, in genie.metaparser._metaparser.MetaParser.parse
  File "src/genie/metaparser/_metaparser.py", line 322, in genie.metaparser._metaparser.MetaParser.parse
  File "src/genie/metaparser/util/schemaengine.py", line 318, in genie.metaparser.util.schemaengine.Schema.validate
  File "src/genie/metaparser/util/schemaengine.py", line 288, in genie.metaparser.util.schemaengine.Schema.validate
  File "src/genie/metaparser/util/schemaengine.py", line 390, in genie.metaparser.util.schemaengine.Schema.validate
  File "src/genie/metaparser/util/schemaengine.py", line 363, in genie.metaparser.util.schemaengine.Schema.validate
  File "src/genie/metaparser/util/schemaengine.py", line 318, in genie.metaparser.util.schemaengine.Schema.validate
  File "src/genie/metaparser/util/schemaengine.py", line 288, in genie.metaparser.util.schemaengine.Schema.validate
  File "src/genie/metaparser/util/schemaengine.py", line 390, in genie.metaparser.util.schemaengine.Schema.validate
  File "src/genie/metaparser/util/schemaengine.py", line 363, in genie.metaparser.util.schemaengine.Schema.validate
  File "src/genie/metaparser/util/schemaengine.py", line 390, in genie.metaparser.util.schemaengine.Schema.validate
  File "src/genie/metaparser/util/schemaengine.py", line 363, in genie.metaparser.util.schemaengine.Schema.validate
  File "src/genie/metaparser/util/schemaengine.py", line 390, in genie.metaparser.util.schemaengine.Schema.validate
  File "src/genie/metaparser/util/schemaengine.py", line 363, in genie.metaparser.util.schemaengine.Schema.validate
  File "src/genie/metaparser/util/schemaengine.py", line 430, in genie.metaparser.util.schemaengine.Schema.validate
genie.metaparser.util.exceptions.SchemaUnsupportedKeyError: bridge_group.VPLS.bridge_domain.TEST.vfi.neighbor: Unsupported keys: ['172.16.154.59', '172.16.154.60']

**Comment by SohanTirpude** [comment]
Hello @pyhas, kindly let us know which version of genieparser you are using. Also, kindly provide the output as well.

**Comment by sowmyadn010501** [comment]
Kindly give us update on this.

**Comment by pyhas** [comment]
Thank you @SohanTirpude for looking into it, I am using genie version 24.5, below is the command output.

```
Bridge group: VPLS, bridge-domain: TEST, id: 0, state: up, ShgId: 0, MSTi: 0
  Aging: 300 s, MAC limit: 64000, Action: none, Notification: syslog
  Filter MAC addresses: 0
  ACs: 1 (1 up), VFIs: 1, PWs: 6 (4 up), PBBs: 0 (0 up), VNIs: 0 (0 up)
  List of ACs:
    BE3, state: up, Static MAC addresses: 0
  List of Access PWs:
      Neighbor 172.16.154.59 pw-id 5966, state: down, Static MAC addresses: 0
      Neighbor 172.16.154.60 pw-id 6066, state: down, Static MAC addresses: 0
  List of VFIs:
    VFI TEST (up)
      Neighbor 172.16.152.13 pw-id 20161:10090, state: up, Static MAC addresses: 0
      Neighbor 172.16.152.67 pw-id 20161:10090, state: up, Static MAC addresses: 0
      Neighbor 172.16.154.190 pw-id 20161:10090, state: up, Static MAC addresses: 0
      Neighbor 172.16.154.191 pw-id 20161:10090, state: up, Static MAC addresses: 0

```

**Comment by sowmyadn010501** [comment]
thanks for the output. Kindly give us some time

**Comment by SohanTirpude** [comment]
Hello @pyhas,

We have found the issue and solution as well. Now, with solution, this is how the output looks like:
```
{
    "bridge_group": {
        "VPLS": {
            "bridge_domain": {
                "TEST": {
                    "id": 0,
                    "state": "up",
                    "shg_id": 0,
                    "mst_i": 0,
                    "mac_aging_time": 300,
                    "mac_limit": 64000,
                    "mac_limit_action": "none",
                    "mac_limit_notification": "syslog",
                    "filter_mac_address": 0,
                    "ac": {
                        "num_ac": 1,
                        "num_ac_up": 1,
                        "interfaces": {
                            "Bundle-Ether3": {
                                "state": "up",
                                "static_mac_address": 0
                            }
                        }
                    },
                    "vfi": {
                        "num_vfi": 1,
                        "TEST": {
                            "state": "up",
                            "neighbor": {
                                "172.16.152.13": {
                                    "pw_id": {
                                        "20161:10090": {
                                            "state": "up",
                                            "static_mac_address": 0
                                        }
                                    }
                                },
                                "172.16.152.67": {
                                    "pw_id": {
                                        "20161:10090": {
                                            "state": "up",
                                            "static_mac_address": 0
                                        }
                                    }
                                },
                                "172.16.154.190": {
                                    "pw_id": {
                                        "20161:10090": {
                                            "state": "up",
                                            "static_mac_address": 0
                                        }
                                    }
                                },
                                "172.16.154.191": {
                                    "pw_id": {
                                        "20161:10090": {
                                            "state": "up",
                                            "static_mac_address": 0
                                        }
                                    }
                                }
                            }
                        }
                    },
                    "pw": {
                        "num_pw": 6,
                        "num_pw_up": 4,
                        "neighbor": {
                            "172.16.154.59": {
                                "pw_id": {
                                    "5966": {
                                        "state": "down",
                                        "static_mac_address": 0
                                    }
                                }
                            },
                            "172.16.154.60": {
                                "pw_id": {
                                    "6066": {
                                        "state": "down",
                                        "static_mac_address": 0
                                    }
                                }
                            }
                        }
                    },
                    "pbb": {
                        "num_pbb": 0,
                        "num_pbb_up": 0
                    },
                    "vni": {
                        "num_vni": 0,
                        "num_vni_up": 0
                    }
                }
            }
        }
    }
}
```
Kindly verify if everything is correct. 

Thank you.

**Comment by pyhas** [comment]
still having the same error.

**Comment by SohanTirpude** [comment]
Hello @pyhas, 

The fix has not merged yet. I just raised the PR for this. One it is merged, I will let you know. 

Meanwhile, kindly verify above shared output and let us know if anything is wrong in it.

Thank you.

**Comment by SohanTirpude** [comment]
Hello @pyhas,

The PR has been merged and the fix will be available in the upcoming pyats 24.6 release. The one thing is to use this fix you will have to use the revision number 1, like this:
```
device.parse("show l2vpn bridge-domain", revision=1)
```
You can find more information in this document: https://pubhub.devnetcloud.com/media/pyats-development-guide/docs/writeparser/writeparser.html#revising-a-parser

Meanwhile, I am closing this ticket. If you are facing any other issues, kindly open a new ticket.

Thank you.

**Comment by pyhas** [comment]
Thank you @SohanTirpude.
